### PR TITLE
Remove internally generated DataStore fields

### PR DESCRIPTION
**Pull request by anuveyatsu**
Such as `_id`, `_full_text` and `_count`.

### CODE REVIEW

File: lib/dms.js [hunk 1]
```diff
@@ -32,8 +32,8 @@ class DmsModel {
   async getPackage(name) {
     const action = 'package_show'
     let url = new URL(resolve(this.api, action))
-    url.search = `name_or_id=${name}`
-    
+    url.search = `id=${name}`
```
[the remark this answers is not in the document]
Comment (anuveyatsu) [reply lib/dms.js]: I've checked with 2 instances (Energinet and National Grid) and worked just fine. Actually, `name_or_id` was breaking Energinet after some backend updates. Also, see official docs:

https://docs.ckan.org/en/2.8/api/index.html#ckan.logic.action.get.package_show

[the remark this answers is not in the document]
Comment (anuveyatsu) [reply lib/dms.js]: @starsinmypockets

File: lib/dms.js [hunk 2]
```diff
@@ -75,10 +75,10 @@ class DmsModel {
         throw response
       }
       response = await response.json()
-      const { fields } = response.result
-      for (let i = 0; i < fields.length; i++) {
-        fields[i] = utils.dataStoreDataDictionaryToTableSchema(fields[i])
-      }
+      const fields = response.result.fields
+        .map(field => utils.dataStoreDataDictionaryToTableSchema(field))
+        .filter(field => field)
```
[the remark this answers is not in the document]
Comment (anuveyatsu) [reply lib/dms.js]: @starsinmypockets yes, exactly. `utils.dataStoreDataDictionaryToTableSchema` returns `null` if a field is DataStore generated internal field. See:

https://github.com/datopian/frontend-v2/pull/149/commits/709e081b86b15e0bfa1bea1cf63b0465cd9c8222

**Comment by starsinmypockets** [comment]
Approved ;)

**Pull request merged**